osu.Game/Overlays/OverlayColourProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayColourProvider
+    {
+        private readonly OverlayColourScheme colourScheme;
+
+        public OverlayColourProvider(OverlayColourScheme colourScheme)
+        {
+            this.colourScheme = colourScheme;
+        }
+
+        private Color4 convert(float saturation, float lightness) => Color4.FromHsl(new Vector4(getBaseHue(colourScheme), saturation, lightness, 1));
+
+        // See https://github.com/ppy/osu-web/blob/4218c288292d7c810b619075471eaea8bbb8f9d8/app/helpers.php#L1463
+        private static float getBaseHue(OverlayColourScheme colourScheme)
+        {
+            float hue;
+
+            switch (colourScheme)
+            {
+                default:
+                    throw new ArgumentException($@"{colourScheme} colour scheme does not provide a hue value in {nameof(getBaseHue)}.");
+
+                case OverlayColourScheme.Red:
+                    hue = 0;
```

Comment:
just change these to `return` statements and inline the `/ 360f`. i think it will be easier to read.